pkg/audit/manager.go

```diff
@@ -274,62 +275,107 @@ func (am *Manager) auditResources(
 	var errs opa.Errors
 	nsCache := newNSCache()
 
-	for gv, gvKinds := range clusterAPIResources {
-		for kind := range gvKinds {
-			objList := &unstructured.UnstructuredList{}
-			opts := &client.ListOptions{
-				Limit: int64(*auditChunkSize),
+	constraintsList := make(map[string]bool)
+	if *auditMatchKindOnly {
+		constraintList := &unstructured.UnstructuredList{}
+		for _, c := range resourceList {
+			constraintList.SetGroupVersionKind(c)
+			if err = am.client.List(ctx, constraintList); err != nil {
+				am.log.Error(err, "Unable to list objects for gvk", "group", c.Group, "version", c.Version, "kind", c.Kind)
+				continue
 			}
-			resourceVersion := ""
-
-			for {
-				objList.SetGroupVersionKind(schema.GroupVersionKind{
-					Group:   gv.Group,
-					Version: gv.Version,
-					Kind:    kind + "List",
-				})
-				objList.SetResourceVersion(resourceVersion)
-
-				err := am.client.List(ctx, objList, opts)
+			for _, constraint := range constraintList.Items {
+				kinds, found, err := unstructured.NestedSlice(constraint.Object, "spec", "match", "kinds")
 				if err != nil {
-					am.log.Error(err, "Unable to list objects for gvk", "group", gv.Group, "version", gv.Version, "kind", kind)
+					am.log.Error(err, "Unable to return spec.match.kinds field", "group", c.Group, "version", c.Version, "kind", c.Kind)
 					continue
 				}
+				if found {
+					for _, k := range kinds {
+						kind, ok := k.(map[string]interface{})
+						if !ok {
+							am.log.Error(errors.New("could not cast kind as map[string]"), "kind", k)
+							continue
+						}
+						kindsKind, _, err := unstructured.NestedSlice(kind, "kinds")
+						if err != nil {
+							am.log.Error(err, "Unable to return kinds.kinds field", "group", c.Group, "version", c.Version, "kind", c.Kind)
+							continue
+						}
+						for _, kk := range kindsKind {
+							// adding constraint match kind to list
+							constraintsList[kk.(string)] = true
+						}
+					}
+				} else {
+					// if constraint doesn't have match kinds defined, we will look at all kinds
+					constraintsList["*"] = true
+				}
+			}
+		}
+	} else {
+		constraintsList["*"] = true
+	}
 
-				for _, obj := range objList.Items {
-					objNamespace := obj.GetNamespace()
-					if am.skipExcludedNamespace(objNamespace) {
+	for gv, gvKinds := range clusterAPIResources {
+		for kind := range gvKinds {
+			_, found := constraintsList[kind]
+			if found || constraintsList["*"] {
+				objList := &unstructured.UnstructuredList{}
+				opts := &client.ListOptions{
+					Limit: int64(*auditChunkSize),
+				}
+				resourceVersion := ""
+
+				for {
```

Comment:
I'm guessing the only dfference in this inner for-loop is indentation level?

Reply:
that's right, no changes here